pkg/web/api/connector_v1_test.go

```diff
@@ -284,6 +284,39 @@ func TestConnectorAPIv1_DeleteConnector(t *testing.T) {
 	assert.Equal(t, want, got)
 }
 
+func TestConnectorAPIv1_ValidateConnector(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	csMock := apimock.NewConnectorOrchestrator(ctrl)
+	api := NewConnectorAPIv1(csMock)
+	connBuilder := connmock.Builder{Ctrl: ctrl}
+
+	source := connBuilder.NewSourceMock(uuid.NewString(), connector.Config{
+		Name:     "A source connector",
+		Settings: map[string]string{"path": "path/to"},
+		Plugin:   "builtin:file",
+	})
```

Comment:
Do we actually need to build the source? It would feel a bit more natural without building, since we will always first validate a config, and then build a source. And also, it looks like it acts more like a container: we only use it to get the type and config. In other words, it might make sense to just declare the name, settings and plugin variables and use them directly.